packages/cli/config/jest.js

```diff
@@ -25,6 +25,50 @@ async function getConfig() {
     return require(path.resolve('jest.config.ts'));
   }
 
+  // We read all "jest" config fields in package.json files all the way to the filesystem root.
+  // All configs are merged together to create the final config, with longer paths taking precedence.
+  // The merging of the configs is shallow, meaning e.g. all transforms are replaced if new ones are defined.
+  const pkgJsonConfigs = [];
+  let currentPath = process.cwd();
+
+  // Some sanity check to avoid infinite loop
+  for (let i = 0; i < 100; i++) {
+    const packagePath = path.resolve(currentPath, 'package.json');
+    const exists = fs.pathExistsSync(packagePath);
+    if (exists) {
+      try {
+        const data = fs.readJsonSync(packagePath);
+        if (data.jest) {
+          pkgJsonConfigs.unshift(data.jest);
+        }
+      } catch (error) {
+        throw new Error(
+          `Failed to parse package.json file reading jest configs, ${error}`,
+        );
+      }
+    }
+
+    const newPath = path.dirname(currentPath);
+    if (newPath === currentPath) {
+      break;
+    }
+    currentPath = newPath;
+  }
+
+  // We add an additional Jest config parameter only known by the Backstage CLI
+  // called `transformModules`. It's a list of modules that we want to apply
+  // our configured jest transformations for.
+  // This is useful when packages are published in untranspiled ESM or TS form.
+  const transformModules = pkgJsonConfigs
+    .flatMap(conf => {
+      const modules = conf.transformModules || [];
+      delete conf.transformModules;
+      return modules;
+    })
+    .map(name => `${name}/`)
+    .join('|');
+  const transformModulePattern = transformModules && `(?!${transformModules})`;
```

Comment:
Hehe, the `&&` is fancy, but maybe a ternary would be clearer :)

Reply:
Meeeeh :p